File: java/src/main/java/fr/sdecout/handson/persistence/author/AuthorRepository.java
package fr.sdecout.handson.persistence.author;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.stream.Stream;

public interface AuthorRepository extends JpaRepository<AuthorEntity, String> {
    Stream<AuthorEntity> findByLastNameLikeIgnoringCase(String hint);
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/persistence/author/AuthorRepository.kt
package fr.sdecout.handson.persistence.author

import org.springframework.data.jpa.repository.JpaRepository

@Deprecated("Replace with jOOQ")
interface AuthorRepository : JpaRepository<AuthorEntity, String> {
    fun findByLastNameLikeIgnoringCase(hint: String): List<AuthorEntity>
}
